### PR TITLE
Integrare EVie in Jersey

### PR DESCRIPTION
**Pull request by Stefanxdan**
no authorization
docked: true

Nu primesc pretul de la provider asa ca m-am uitat de pe site ul lor si am hardocat pretul in functie de modelul masinii
model_id -> model -> price_string
378 -> Nissan E-NV200 -> 11 / hour or 60 / day
286 -> BMW i3 -> 9 / hour or 60 / day
381 -> Nissan Leaf -> 7.5 / hour or 50 / day

INSERT INTO `urbanair-integrations`.`region` (`uniqueName`, `radius`, `center`) VALUES ('St Helier, Jersey', '3', '{\"latitude\": 49.191713, \"longitude\": -2.103390}');
INSERT INTO `urbanair-integrations`.`providers` (`providerId`, `type`, `ctaColor`, `ctaBgColor`, `iosDeeplinks`, `andDeeplinks`, `active`, `supported`, `metadata`, `crawlable`, `rideType`, `name`) VALUES ('yr08c7Evie', 'evie', '#FFFFFF', '#1963ff', '{\"store\": \"https://apps.apple.com/tt/app/evie-car-share/id1479674635\"}', '{\"store\": \"https://play.google.com/store/apps/details?id=com.goodtravelsoftware.sharecar.evie&hl=en\"}', '1', '1', '{\"assets\":{\"url\":\"https://api.share.car/v2/stations-explore?scheme_key=evie\",\"method\":\"GET\",\"headers\":{\"Accept\":\"application/json\",\"Host\":\"api.share.car\"}}}', '1', 'hail', 'EVie');

### CODE REVIEW

File: src/api/provider/providers/evie/EvieProvider.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+                const assetLocation = { latitude, longitude };
+                const battery = AssetChargeLevel.validFromNumber(batteryLevel);
+
+                var pricePerHour : number = 0;
+                var pricePerDay : number;
+                switch(modelId){
+                    case 378:
+                        pricePerHour = 11;
+                        pricePerDay = 60;
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/evie/EvieProvider.ts:84]: trebuie sa pui `break;` dupa fiecare use case, altfel o sa iti treaca prin toate care sunt sub cazul in care te afli tu

File: src/api/provider/providers/evie/EvieProvider.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    case 286:
+                        pricePerHour = 9;
+                        pricePerDay = 60;
+                    case 381:
+                        pricePerHour = 7.5;
+                        pricePerDay = 50;
+                }
+                var price: string;
+                if(pricePerHour != 0)
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/evie/EvieProvider.ts:93]: incearca sa folosesti inegalitatea stricta `!==`

File: src/api/provider/providers/evie/EvieProvider.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                        pricePerHour = 11;
+                        pricePerDay = 60;
+                    case 286:
+                        pricePerHour = 9;
+                        pricePerDay = 60;
+                    case 381:
+                        pricePerHour = 7.5;
+                        pricePerDay = 50;
+                }
+                var price: string;
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/evie/EvieProvider.ts:92]: (asta e doar un detaliu, nu e nimic gresit in cod) din cauza comportamentului variabilelor declarate cu `var` (se pot intampla lucruri ciudate pentru ca se face hoisting) se recomanda sa folosesti `let`

File: src/api/provider/providers/evie/EvieProvider.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                if(pricePerHour != 0)
+                    price = pricePerHour.toString() + ' £ / hour or ' + pricePerDay.toString() + " £ / day" ;
+                else
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/evie/EvieProvider.ts:95]: puteai sa rezolvi cazul asta mai simplu daca puneai un `default:` case in switch-ul de mai sus

File: src/api/provider/providers/evie/EvieProvider.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                const currency_name = asset?.hourlyCost?.currency_name;
+                const currency_symbol = asset?.hourlyCost?.currency_symbol;
+
+
+                const rawId = vehicle?.id;
+                const name = vehicle?.fullName;
+                const batteryLevel = vehicle?.vitals?.fuelLevel;
+                const modelId = vehicle?.model?.id;
+
+                const imageUrl = input?.providerGenericAssetImage;
```
Comment (vladgaujaneanu24) [inline src/api/provider/providers/evie/EvieProvider.ts:68]: `const imageUrl = ProviderType.assetModelUrl(providerId, modelId.toString());`

Comment (Stefanxdan) [reply src/api/provider/providers/evie/EvieProvider.ts:68]: Eu de la Evi primesc mai multe detalii despre model:
"model": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; "id": 378,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"name": "E-NV200",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"fullName": "NISSAN E-NV200",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"vehicleType": {... },
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"imageFilename": "https://images.share.car/vehicle-models/1279302412-nissan-e-nv200.png"
...
}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Ce ar trebuie sa pun:
const imageUrl = ProviderType.assetModelUrl(providerId, modelId.toString());
const imageUrl = ProviderType.assetModelUrl(providerId, name);
const imageUrl = ProviderType.assetModelUrl(providerId, fullName.replace(" ","_") );
const imageUrl = ProviderType.assetModelUrl(providerId, imageFilename);
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;?

Comment (vladgaujaneanu24) [reply src/api/provider/providers/evie/EvieProvider.ts]: pune `modelId`. Importatnt e sa avem un identificator unic pe model, de orice fel ar fi el

**Pull request merged**